src/routes/v2/projects.rs

```diff
@@ -2558,3 +2561,134 @@ pub async fn delete_from_index(
+        return Ok(HttpResponse::NotFound().body(""));
+    }
+    let mut members_data =
+        db_models::TeamMember::get_from_team_full(project_data.inner.team_id, &**pool, &redis)
+            .await?;
+    let mut member_user_ids = members_data.iter().map(|x| x.user_id).collect::<Vec<_>>();
+    if let Some(oid) = project_data.inner.organization_id {
+        let organization_data = db_models::Organization::get_id(oid, &**pool, &redis).await?;
+        if let Some(organization_data) = organization_data {
+            let org_team = db_models::TeamMember::get_from_team_full(
+                organization_data.team_id,
+                &**pool,
+                &redis,
+            )
+            .await?;
+            for member in org_team {
+                if !member_user_ids.contains(&member.user_id) {
+                    member_user_ids.push(member.user_id);
+                    members_data.push(member);
+                }
+            }
+        }
+    }
+    let users = db_models::User::get_many_ids(&member_user_ids, &**pool, &redis).await?;
+    let versions = db_models::Version::get_many(&project_data.versions, &**pool, &redis).await?;
+    let mut versions = filter_authorized_versions(versions, &None, &pool).await?;
+    versions.sort_by(|a, b| b.date_published.cmp(&a.date_published));
+    let authors = users
+        .into_iter()
+        .map(|u| atom_syndication::Person {
+            uri: Some(format!("{site_url}/user/{}", &u.username)),
+            name: u.username,
+            email: None,
+        })
+        .collect::<Vec<_>>();
+
+    let project = models::projects::Project::from(project_data);
+    let mut feed = atom_syndication::Feed::default();
+    feed.title = atom_syndication::Text::plain(project.title.to_string());
+    feed.links.push(atom_syndication::Link {
+        href: format!("{site_url}/mod/{string}"),
+        rel: "alternate".to_string(),
+        mime_type: Some("text/html".to_string()),
+        title: Some(project.title.to_string()),
+        hreflang: None,
+        length: None,
+    });
+    feed.links.push(atom_syndication::Link {
+        href: format!("{self_addr}/project/{string}/feed"),
+        rel: "self".to_string(),
+        mime_type: Some("application/atom+xml".to_string()),
+        title: Some(project.title.to_string()),
+        hreflang: None,
+        length: None,
+    });
+    feed.icon = project.icon_url;
+    feed.id = format!("{site_url}/mod/{string}");
+    feed.updated = project.updated.fixed_offset();
+    feed.rights = Some(project.license.name.into());
+    for category in &project.categories {
+        feed.categories.push(atom_syndication::Category {
+            term: category.to_string(),
+            label: Some(category.to_string()),
+            scheme: None,
+        });
+    }
+    for version in versions {
+        let mut entry = atom_syndication::Entry::default();
+        let link = format!("{site_url}/mod/{string}/version/{}", version.version_number);
```

Comment:
Versions can have identical version numbers, so this would lead to conflicting IDs.

Reply:
Fixed. It now only links to version numbers if there's no conflict, and falls back to version IDs. The entry ID always uses the permanent version ID (and project ID).